### PR TITLE
Print recipe signature if missing arguments

### PR DESCRIPTION
**Pull request by ladysamantha**
references issue #341 

Wasn't sure what format would be the best, but I took a stab at it and and printed out a simple usage doc on top of the error that was already there. Let me know what more I can do to help/make the code cleaner! I'm still fairly new to rust, so if this seems like a naÏve approach, please forgive me :laughing:

### CODE REVIEW

**Comment by casey** [comment]
Thank you for the PR! I think this has a stray 2nd commit. You might want to create a separate branch for the pull request, so that pushes to master don't land on the PR after you've opened it.

File: src/runtime_error.rs [hunk 1]
```diff
@@ -71,12 +71,13 @@ impl<'a> Display for RuntimeError<'a> {
...
       },
-      ArgumentCountMismatch{recipe, found, min, max} => {
+      ArgumentCountMismatch{recipe, ref parameters, found, min, max} => {
         if min == max {
           let expected = min;
-          write!(f, "Recipe `{}` got {} argument{} but {}takes {}",
+          write!(f, "Recipe `{}` got {} argument{} but {}takes {}\n",
                     recipe, found, maybe_s(found),
                     if expected < found { "only " } else { "" }, expected)?;
+          write!(f, "Usage:\n just {} {}", recipe, parameters.join(" "));
```
Comment (casey) [inline src/runtime_error.rs]: Nit: Usage should be lowercase, since `error:` is also lowercase.

**Review by casey — CHANGES_REQUESTED** [review]
Parameters have a `Display` implementation, so that should be reused to format optional and 
 variadic parameters, and to color the usage string. I use the `{:#}` formatter for colored output. Since this message is being printed to stderr, it should use `{:#}` of stderr is attached to a terminal, and `{}` otherwise. Check out `run.rs` for how to do this.

File: src/runtime_error.rs (hunk 1 above)
Comment (casey) [reply src/runtime_error.rs]: Also, let's go with four spaces of indentation in front of the word `just`.

File: src/justfile.rs [hunk 2]
```diff
@@ -308,13 +308,12 @@ a return code:
...
       ArgumentCountMismatch {
         recipe,
-        parameters,
+        parameters: _,
         found,
         min,
         max,
       } => {
         assert_eq!(recipe, "a");
-        assert_eq!(parameters, ["b", "c", "d"]);
```
Comment (casey) [inline src/justfile.rs]: For the tests, let's leave in the assertion that the names are right. You can collect into a vec of the recipe names and compare with that.

**Review by casey — CHANGES_REQUESTED** [review]

**Comment by ladysamantha** [comment]
> Thank you for the PR! I think this has a stray 2nd commit. You might want to create a separate branch for the pull request, so that pushes to master don't land on the PR after you've opened it.

D'oh! That actually makes sense, I'm just so used to not using branches at work (company policy...) so I forget sometimes to use them lol

Did you want the usage printed on all of the conditions in `ArugmentCountMismatch`?

**Review by casey — CHANGES_REQUESTED** [review]
Sweet, thanks for making those changes. We should check stderr instead of stdout, and I think we should print the usage string in all cases. After that this should be good to go.

**Comment by casey** [comment]
Awesome, thank for your contribution!

**Comment by ladysamantha** [comment]
Thank you! Happy to contribute :blue_heart:

**Pull request merged**